src/debugAdapter/goDebug.ts

```diff
@@ -751,6 +752,23 @@ class GoDebugSession extends LoggingDebugSession {
 		});
 	}
 
+	protected setBreakPointsRequest(response: DebugProtocol.SetBreakpointsResponse, args: DebugProtocol.SetBreakpointsArguments): void {
+		log('SetBreakPointsRequest');
+		if (!this.continueRequestRunning) {
+			this.setBreakPoints(response, args);
+		} else {
+			this.skipStopEventOnce = true;
+			this.delve.callPromise('Command', [{ name: 'halt' }]).then(() => {
+				return this.setBreakPoints(response, args);
+			}, err => {
+				this.sendErrorResponse(response, 2002, 'Failed to set breakpoint: "{e}"', { e: err.toString() });
```

Comment:
`Failed to halt delve before attempting to set breakpoint` would be more informative

Reply:
Updated